futurex_openedx_extensions/helpers/tenants.py

```diff
@@ -314,3 +316,61 @@ def get_tenants_sites(tenant_ids: List[int]) -> List[str]:
 def get_nafath_sites() -> List:
     """Get all nafath sites"""
     return get_all_tenants_info()['special_info']['nafath_sites']
+
+
+def generate_tenant_config(sub_domain: str, platform_name: str) -> dict:
+    """
+    Generate a tenant configuration by copying the default config and replacing placeholders.
+
+    :param sub_domain: The subdomain to be used for the tenant
+    :param platform_name: The platform name to be used for the tenant
+    :return: Updated tenant configuration with placeholders replaced
+    :rtype: dict
+    """
+    try:
+        default_config = TenantConfig.objects.get(route__domain=settings.FX_DEFAULT_TENANT_SITE)
+        config_lms_dict = json.dumps(default_config.lms_configs)
+        config_lms_dict = config_lms_dict.replace('{{platform_name}}', platform_name)
+        config_lms_dict = config_lms_dict.replace('{{sub_domain}}', sub_domain)
+        return json.loads(config_lms_dict)
+    except TenantConfig.DoesNotExist as exc:
+        raise FXCodedException(
+            code=FXExceptionCodes.TENANT_NOT_FOUND,
+            message=f'Default TenantConfig not found! default site: ({settings.FX_DEFAULT_TENANT_SITE})',
+        ) from exc
+
+
+def create_new_tenant_config(sub_domain: str, platform_name: str) -> TenantConfig:
+    """
+    Creates a new TenantConfig and associated Route with the provided subdomain and domain name.
+
+    :param sub_domain: The subdomain to be used for the tenant
+    :param platform_name: The platform name to be used for the tenant
+    :return: The created TenantConfig object
+    """
+    site_domain = f'{sub_domain}.{settings.FX_TENANTS_BASE_DOMAIN}'
+    if Route.objects.filter(domain=site_domain).exists():
+        raise FXCodedException(
+            code=FXExceptionCodes.ROUTE_ALREADY_EXIST,
+            message=f'Route already exists with site domain: ({site_domain}).',
+        )
+    if TenantConfig.objects.filter(external_key=sub_domain).exists():
+        raise FXCodedException(
+            code=FXExceptionCodes.TENANT_ALREADY_EXIST,
+            message=f'Tenant already exists for given sub_domain: ({sub_domain}).',
+        )
```

Comment:
No need. The external key is a random number sometimes. We'll trust `Route` and that's it
```suggestion
```